journalism_workflow/adjust_photos.py

```diff
@@ -0,0 +1,91 @@
+import os
+import tempfile
+
+from django.conf import settings
+from PIL import Image
+
+from orchestra.google_apps.convenience import create_folder_with_permissions
+from orchestra.google_apps.permissions import write_with_link_permission
+from orchestra.google_apps.service import Service
+
+import logging
+logger = logging.getLogger(__name__)
+
+
+def autoadjust_photos(project_data, prerequisites):
+    """Resize all images in a google drive directory."""
+    print(project_data)
+    print(prerequisites)
+    task_data = {}
+    parent_folder_id = project_data['project_folder_id']
+
+    # Create a directory to output the photos
+    output_folder = create_folder_with_permissions(
+        parent_folder_id,
+        'Processed Photos',
+        permissions=[write_with_link_permission],
+    )
+    task_data['processed_photo_folder'] = output_folder['id']
+
+    # List the existing photos
+    raw_photo_folder_id = (prerequisites
+                           .get('photographer')
+                           .get('prerequisites')
+                           .get('editor')
+                           .get('prerequisites')
+                           .get('document_creation')
+                           .get('task')
+                           .get('data')
+                           .get('raw_photo_folder'))
+    service = Service(settings.GOOGLE_P12_PATH,
+                      settings.GOOGLE_SERVICE_EMAIL)
+    photos_metadata = service.list_folder(raw_photo_folder_id)
+
+    # Iterate over the input photos and process them.
+    task_data['photos_for_caption'] = []
+    for photo_metadata in photos_metadata:
+        photo, title, mimetype = download_image(service, photo_metadata)
+        upload = upload_image(service,
+                              task_data['processed_photo_folder'],
+                              adjust_photo(photo),
+                              title,
+                              mimetype)
+        embed_link = upload['webContentLink'].replace('&export=download', '')
+        task_data['photos_for_caption'].append(embed_link)
+    return task_data
+
+
+def adjust_photo(photo):
+
+    # Write the photo to a temporary file
+    temp = tempfile.NamedTemporaryFile(mode='wb', delete=False)
+    temp.write(photo)
+
+    # Open it up and play with it in PIL
+    im = Image.open(temp.name)
+    im = im.convert('L')  # convert to greyscale
+
+    # Save it back out into a new temporary file
+    os.unlink(temp.name)
+    temp2 = tempfile.NamedTemporaryFile(mode='wb', delete=False)
+    im.save(temp2, format='jpeg')
+    return temp2
+
+
+def download_image(service, photo_metadata):
```

Comment:
Conversations with @jrbotros led me to believe that we're trying to move away from google apps dependence, so I didn't want to build up the functionality too heavily.

Happy to create an issue if you think it's worth it.